packages/workers-shared/asset-server-worker/src/index.ts

```diff
@@ -8,48 +9,52 @@ import {
 import { getAdditionalHeaders, getMergedHeaders } from "./utils/headers";
 import { getAssetWithMetadataFromKV } from "./utils/kv";
 
-export default {
-	async fetch(request: Request, env: Env) {
+export default class extends WorkerEntrypoint<Env> {
+	async fetch(request: Request) {
 		if (!request.method.match(/^(get)$/i)) {
 			return new MethodNotAllowedResponse();
 		}
 
 		try {
-			return this.handleRequest(request, env);
+			return this.handleRequest(request);
 		} catch (err) {
 			return new InternalServerErrorResponse(err);
 		}
-	},
+	}
 
-	async handleRequest(request: Request, env: Env) {
-		const {
-			// ASSETS_MANIFEST is a pipeline binding to an ArrayBuffer containing the
-			// binary-encoded site manifest
-			ASSETS_MANIFEST = new ArrayBuffer(0),
-
-			// ASSETS_KV_NAMESPACE is a pipeline binding to the KV namespace that the
-			// assets are in.
-			ASSETS_KV_NAMESPACE,
-		} = env;
+	async hasResponseForRequest(request: Request) {
```

Comment:
`router-worker` will be able to ask `env.assets.hasResponseForRequest(request)` if we want to now, rather than doing the `response.statusCode != 404` sort of thing.